### PR TITLE
Make changes unifying approach for ROS2 humble and iron.

### PR DESCRIPTION
**Pull request by PawelLiberadzki**
Should allow building demo for both humble and iron.
- add iron parameters for OTTOs  + fixed palletization cmake,
- add script configuring Universal_Robot_Driver submodule for ros2 humble and iron,
- update documentation.

### CODE REVIEW

File: ros2_ws/setup_submodules.bash [hunk 1]
```diff
@@ -0,0 +1,10 @@
+git submodule init
+git submodule update
+
+if [ "$ROS_DISTRO" == "iron" ]
+then
+	cd ./src/Universal_Robots_ROS2_Driver/
+	git checkout 129687763140508b4b5cb497b197ce1057defe88
```
Comment (jhanca-robotecai) [inline ros2_ws/setup_submodules.bash:7]: Should we also check if the checkout was successful? I can imagine the following situation (I had it):
```bash
jhanca@robotec-o3de:~/devroot/projects/ROSCon2023Demo/ros2_ws$ ./setup_submodules.bash
fatal: reference is not a tree: 129687763140508b4b5cb497b197ce1057defe88
Submodules updated for ROS2 iron.
```

```suggestion
	git checkout 129687763140508b4b5cb497b197ce1057defe88
	if [ $? -ne 0 ]
	then
            echo "Failed to checkout ROS 2 iron branch, submodule updated for ROS 2 humble."
            exit 1
        fi
```

**Review by jhanca-robotecai — CHANGES_REQUESTED** [review]
The code works on `humble`, but it does not work for `iron`. It is caused by the fact that the submodule is fetched from the `Robotec` repository and the branch `iron` is simply not there. Either update the `Robotec` fork or switch the submodule:

```bash
jhanca@robotec-o3de:~/devroot/projects/ROSCon2023Demo$ cat .gitmodules 
[submodule "ros2_ws/src/Universal_Robots_ROS2_Driver"]
	path = ros2_ws/src/Universal_Robots_ROS2_Driver
	url = https://github.com/RobotecAI/Universal_Robots_ROS2_Driver.git
[submodule "ros2_ws/src/moveit_task_constructor"]
	path = ros2_ws/src/moveit_task_constructor
	url = https://github.com/ros-planning/moveit_task_constructor.git
jhanca@robotec-o3de:~/devroot/projects/ROSCon2023Demo$ cd ros2_ws/src/Universal_Robots_ROS2_Driver/ ; git branch
* (HEAD detached at 698914b)
  main
```

**Comment by jhanca-robotecai** [comment]
I pushed the `iron` branch to `Robotec` fork of the `UR ROS2 Driver` repository. I can confirm the code works correctly now. 

I also created an issue about submodules: https://github.com/RobotecAI/ROSCon2023Demo/issues/254.

File: ros2_ws/setup_submodules.bash [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+fi
+
+echo "Submodules updated for ROS2 $ROS_DISTRO."
```
[the remark this answers is not in the document]
Comment (PawelLiberadzki) [reply ros2_ws/setup_submodules.bash:12]: Technically we have more than humble and iron, and I assume that they can be sourced too. I modify error message slightly and commit it, thank you!

File: ros2_ws/setup_submodules.bash (hunk 1 above)
Comment (PawelLiberadzki) [reply ros2_ws/setup_submodules.bash:9]: Good point, added!

**Review by jhanca-robotecai — APPROVED** [review]
All my comments were resolved, and the code works now correctly.

**Pull request merged**